From 4b962667364649663139f206d5c9e52b3e063a42 Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Mon, 23 Apr 2018 18:17:44 +0800
Subject: [PATCH 3681/5242] MLK-18101-5 gpu: imx: imx8_prg: Add
 prg_put_auxiliary() helper support

commit  bac7f3e1967f7465cd890f5877198d95f6095fdc from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds prg_put_auxiliary() helper support so that users may
set a particular PRG not serve as an auxiliary one.

Signed-off-by: Yuchou Gan <yuchou.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_prg.c    |    9 +++++++++
 include/video/imx8-prefetch.h |    1 +
 2 files changed, 10 insertions(+)

diff --git a/drivers/gpu/imx/imx8_prg.c b/drivers/gpu/imx/imx8_prg.c
index f558d98..f1db1c1 100644
--- a/drivers/gpu/imx/imx8_prg.c
+++ b/drivers/gpu/imx/imx8_prg.c
@@ -303,6 +303,15 @@ void prg_set_auxiliary(struct prg *prg)
 }
 EXPORT_SYMBOL_GPL(prg_set_auxiliary);
 
+void prg_put_auxiliary(struct prg *prg)
+{
+	if (WARN_ON(!prg))
+		return;
+
+	prg->is_auxiliary = false;
+}
+EXPORT_SYMBOL_GPL(prg_put_auxiliary);
+
 void prg_set_blit(struct prg *prg)
 {
 	if (WARN_ON(!prg))
diff --git a/include/video/imx8-prefetch.h b/include/video/imx8-prefetch.h
index 83769f0..aa1bd1c 100644
--- a/include/video/imx8-prefetch.h
+++ b/include/video/imx8-prefetch.h
@@ -42,6 +42,7 @@ void prg_configure(struct prg *prg, unsigned int width, unsigned int height,
 bool prg_stride_double_check(struct prg *prg,
 			     unsigned int stride, dma_addr_t baddr);
 void prg_set_auxiliary(struct prg *prg);
+void prg_put_auxiliary(struct prg *prg);
 void prg_set_blit(struct prg *prg);
 
 struct dprc;
-- 
1.7.9.5

